### DISCUSSION TITLE
useSWR For CRUD operation

### DISCUSSION THREAD

**Discussion by laurent512** [vercel/swr#2021] [General] [answered]
Hi, 

This is a question before going further with SWR. 
My goal is create a dashboard, that will contain tables, in which the user can perform actions (e.g. update cells, delete rows)

Is useSWR a right candidate for fetching & updating the data ? 
I would like to avoid data refresh when user is interacting with the table in the GUI. 

Thank you

**Comment by swand0g** [comment, marked as answer]
Take a look at disabling [Revalidate on Focus](https://swr.vercel.app/docs/revalidation) in the SWR docs if you want to avoid lots data refresh – you could also look at toggling some more settings through the [options object](https://swr.vercel.app/docs/options#options) (which you can set as global config for all hook calls or pass in to individual hook calls)

In terms of other CRUD operations like `POST` and `PUT` looks like it's going to be in SWR 2.0 and a new hook `useSWRMutation` was added in #1450

Think it's been in 2.0 pre-releases since [v2.0.0-beta.0](https://github.com/vercel/swr/releases/tag/2.0.0-beta.0) if you want to try it out
